packages/flame/lib/src/components/component.dart

```diff
@@ -176,6 +176,20 @@ class Component with Loadable {
     nextParent = component;
   }
 
+  final List<Component> _ancestors = [];
+
+  /// A list containing the current parent and its parent, and so on, until it
+  /// reaches a component without a parent.
+  List<Component> ancestors() {
+    _ancestors.clear();
```

Comment:
Why not just declare the list inside the method?

Reply:
Because then we need to create multiple new objects on every tick when collision detection is used. This has shown to be quite expensive in Forge2D when you get some volume, probably because the GC has to run more often.

Reply:
Oh, that interesting